<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QStegano</class>
 <widget class="QMainWindow" name="QStegano">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>668</width>
    <height>448</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QStegano</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="1" column="0" rowspan="2">
     <widget class="QStackedWidget" name="swInput">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="pSelect">
       <layout class="QGridLayout" name="gridLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>4</number>
        </property>
        <item row="0" column="1">
         <widget class="QPushButton" name="pbSelectFile">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>50</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Select image</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="pPreview">
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>4</number>
        </property>
        <item row="0" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Input image</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="3" column="0">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="pbSelectNewFile">
             <property name="text">
              <string>Select image</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="0" colspan="3">
            <widget class="QLabel" name="imageLabel">
             <property name="minimumSize">
              <size>
               <width>300</width>
               <height>300</height>
              </size>
             </property>
             <property name="text">
              <string>Input   image</string>
             </property>
             <property name="scaledContents">
              <bool>false</bool>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="3">
            <widget class="QLabel" name="imageName">
             <property name="text">
              <string>Input image name</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="3">
            <widget class="QLabel" name="lblSize">
             <property name="text">
              <string>Input image size</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Settings</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="3" column="0" colspan="2">
            <widget class="QStackedWidget" name="swType">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <widget class="QWidget" name="pHide">
              <layout class="QGridLayout" name="gridLayout_7">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <property name="spacing">
                <number>4</number>
               </property>
               <item row="8" column="1">
                <widget class="QPushButton" name="pbSave">
                 <property name="text">
                  <string>Save output image</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QComboBox" name="cbBits">
                 <item>
                  <property name="text">
                   <string>1</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>2</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>4</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>8</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="8" column="0">
                <widget class="QPushButton" name="pbHide">
                 <property name="text">
                  <string>Hide</string>
                 </property>
                </widget>
               </item>
               <item row="7" column="1">
                <widget class="QLabel" name="lblText">
                 <property name="text">
                  <string>---</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Used bits</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="7" column="0">
                <widget class="QLabel" name="label_15">
                 <property name="text">
                  <string>Text length</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QComboBox" name="cbMethod">
                 <item>
                  <property name="text">
                   <string>basic</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>xor</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>inverted</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>opap</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="6" column="0" colspan="2">
                <widget class="QTextEdit" name="teInput">
                 <property name="placeholderText">
                  <string>text to be hidden</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Method</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="pDecode">
              <layout class="QGridLayout" name="gridLayout_8">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <property name="spacing">
                <number>4</number>
               </property>
               <item row="3" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>Used bits</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="5" column="0" colspan="2">
                <widget class="QTextEdit" name="teOutput">
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                 <property name="placeholderText">
                  <string>decoded text</string>
                 </property>
                </widget>
               </item>
               <item row="6" column="0">
                <widget class="QPushButton" name="pbDecode">
                 <property name="text">
                  <string>Decode</string>
                 </property>
                </widget>
               </item>
               <item row="6" column="1">
                <widget class="QPushButton" name="pbCopyDecoded">
                 <property name="text">
                  <string>Copy decoded text</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Method</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLabel" name="lblDecodedMethod">
                 <property name="text">
                  <string>---</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QLabel" name="lblDecodedBits">
                 <property name="text">
                  <string>---</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="cbType">
             <item>
              <property name="text">
               <string>hide</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>decode</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Operation</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>668</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="actionAboutSteganography"/>
    <addaction name="actionAboutApplication"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuAbout"/>
  </widget>
  <action name="actionAboutSteganography">
   <property name="text">
    <string>About steganography</string>
   </property>
  </action>
  <action name="actionAboutApplication">
   <property name="text">
    <string>About application</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>QStegano</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
